openapi-cli/src/main/java/io/ballerina/generators/auth/BallerinaAuthConfigGenerator.java

```diff
@@ -0,0 +1,487 @@
+        }
+        return null;
+    }
+
+    /**
+     * Generate the config parameters of the client class init method.
+     * -- ex: Config param for Http and OAuth 2.0 Authentication mechanisms.
+     *        {@code ClientConfig clientConfig }
+     * -- ex: Config param for API Key Authentication mechanism.
+     *        {@code ApiKeysConfig apiKeyConfig, http:ClientConfiguration clientConfig = {} }
+     * -- ex: Config param for API Key Authentication mechanism.
+     *        {@code http:ClientConfiguration clientConfig = {} }
+     *
+     * @return  {@link List<Node>}  syntax tree node list of config parameters
+     */
+    public static List<Node> getConfigParamForClassInit() {
+        List<Node> parameters  = new ArrayList<>();
+        IdentifierToken equalToken = createIdentifierToken(GeneratorConstants.EQUAL);
+        NodeList<AnnotationNode> annotationNodes = createEmptyNodeList();
+        if (isHttpOROAuth) {
+            BuiltinSimpleNameReferenceNode typeName = createBuiltinSimpleNameReferenceNode(null,
+                    createIdentifierToken(CONFIG_RECORD_NAME));
+            IdentifierToken paramName = createIdentifierToken(CONFIG_RECORD_ARG);
+            RequiredParameterNode authConfig = createRequiredParameterNode(annotationNodes, typeName, paramName);
+            parameters.add(authConfig);
+        } else {
+            if (isAPIKey) {
+                BuiltinSimpleNameReferenceNode apiKeyConfigTypeName = createBuiltinSimpleNameReferenceNode(null,
+                        createIdentifierToken(API_KEY_CONFIG));
+                IdentifierToken apiKeyConfigParamName = createIdentifierToken(API_KEY_CONFIG_PARAM);
+                RequiredParameterNode apiKeyConfigParamNode = createRequiredParameterNode(annotationNodes,
+                        apiKeyConfigTypeName, apiKeyConfigParamName);
+                parameters.add(apiKeyConfigParamNode);
+                parameters.add(createToken(COMMA_TOKEN));
+            }
+            BuiltinSimpleNameReferenceNode httpClientConfigTypeName = createBuiltinSimpleNameReferenceNode(null,
+                    createIdentifierToken("http:ClientConfiguration"));
+            IdentifierToken httpClientConfig = createIdentifierToken(CONFIG_RECORD_ARG);
+            BasicLiteralNode emptyexpression = createBasicLiteralNode(null, createIdentifierToken(" {}"));
+            DefaultableParameterNode defaultHTTPConfig = createDefaultableParameterNode(annotationNodes,
+                    httpClientConfigTypeName,
+                    httpClientConfig, equalToken, emptyexpression);
+            parameters.add(defaultHTTPConfig);
+        }
+        return parameters;
+    }
+
+    /**
+     * Generate {@code http:ClientSecureSocket secureSocketConfig } local variable.
+     *
+     * @return  {@link VariableDeclarationNode} syntax tree variable declaration node.
+     */
+    public static VariableDeclarationNode getSecureSocketInitNode () {
+        if (isHttpOROAuth) {
```

Comment:
@indikasampath2000  The flag `isHttpOROAuth` cannot be removed easily since it has been used in few places like this. Here if we changed it to `!apiKey` it will also include the scenario where there is no security scheme or unsupported security scheme.

Reply:
Ack